src/iuse.cpp

```diff
@@ -141,14 +141,12 @@ void remove_double_ammo_mod( item &it, player &p )
     }
     p.add_msg_if_player( _( "You remove the double battery capacity mod of your %s!" ),
                          it.tname().c_str() );
-    item mod( "battery_compartment", calendar::turn );
+    item mod( "battery_compartment" );
     p.i_add_or_drop( mod, 1 );
     it.item_tags.erase( "DOUBLE_AMMO" );
     // Easier to remove all batteries than to check for the actual real maximum
     if( it.charges > 0 ) {
-        item batteries( "battery", calendar::turn );
-        batteries.charges = it.charges;
-        it.charges = 0;
+        item batteries( "battery", calendar::turn, it.charges );
         p.i_add_or_drop( batteries, 1 );
```

Comment:
Same would apply here. Could also check for `ammo_remaining()` instead of manipulating `charges`. You still need the call to set `it.charges` but this could also be refactored as `ammo_unset`.

End result would be:
```
if( it.ammo_remaining() ) {
    p.i_add_or_drop( { "battery", calendar::turn, it.charges } );
    it.ammo_unset();
}